src/main.rs

```diff
@@ -1,71 +1,106 @@
-use std::collections::VecDeque;
+use logstreamer::{Action, ActionMessage, Response, ResponseMessage};
 use std::io::prelude::*;
 use std::net::{TcpListener, TcpStream};
 use std::sync::{Arc, Mutex};
 use std::thread;
 
-const C_KEY: u8 = 99;
-const P_KEY: u8 = 112;
-const Q_KEY: u8 = 113;
+struct Storage {
+    queue: Mutex<Vec<String>>,
+}
+
+impl Storage {
+    fn new() -> Storage {
+        Storage {
+            queue: Mutex::new(Vec::new()),
+        }
+    }
+
+    fn add_content(&self, content: String) -> u32 {
+        let mut locked_queue = self.queue.lock().unwrap();
+        locked_queue.push(content);
+        (locked_queue.len() - 1) as u32
```

Comment:
The idea here is to return the offset right?
Hummm I might be getting trigger happy with typedefs but something like that may help
```
type Offset = u32

fn add_content(&self, content: String) -> Offset
```
Reminds me of khorikov's course.

Reply:
Now I get it. It's in the Response enum! Does it make sense to add it to the type signature of other methods?
```
    Offset(u32),
```